### PR TITLE
Add search component, memorial index, person index

### PR DESCRIPTION
**Pull request by bashout**
Looking for some feedback on general search structure. Open to changing how it's setup.

- Basic working search of memorials and people. No advanced filters implemented yet.
- Memorial list and details implemented.
- Person list implemented
- Updated editor config to have 4 space indents rather than 2.

### CODE REVIEW

File: src/RollOfHonour.Core/Models/Search/SearchQuery.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using RollOfHonour.Core.Enums;
+
+namespace RollOfHonour.Core.Models.Search;
+
+public class SearchQuery
```
Comment (DavidColemanUK) [inline src/RollOfHonour.Core/Models/Search/SearchQuery.cs]: This class seems a little  unsure of itself.  It has a QueryType, which could be memorial.  At that point PersonType do you feel Person Type is appropriate?  Could more specific types to this job more purposefully?

File: src/RollOfHonour.Data/Repositories/MemorialRepository.cs [hunk 2]
```diff
@@ -1,43 +1,65 @@
...
+        var dbMemorials = _dbContext.WarMemorials.Where(m =>
+            m.Name.Contains(nameFragment))
+            .AsNoTracking();
+
+        if (dbMemorials.Count() == 0)
+        {
+            return null;
+        }
+
+        var results = await dbMemorials.Take(25).Select(m => new MemorialSearchResult() { Id = m.Id, Name = m.Name, Description = m.Description }).ToListAsync();
```
Comment (DavidColemanUK) [inline src/RollOfHonour.Data/Repositories/MemorialRepository.cs]: Reminder we need to do paging at some point.

**Review by DavidColemanUK — COMMENTED** [review]
Looks like a Solid first pass, which was our objective.

File: src/RollOfHonour.Core/Models/Search/SearchQuery.cs (hunk 1 above)
Comment (bashout) [reply src/RollOfHonour.Core/Models/Search/SearchQuery.cs]: This has been done.

File: src/RollOfHonour.Data/Repositories/MemorialRepository.cs (hunk 2 above)
Comment (bashout) [reply src/RollOfHonour.Data/Repositories/MemorialRepository.cs]: 👍

**Review by DavidColemanUK — DISMISSED** [review]
Nice tweak

**Pull request merged**